cpp/src/DataStorm/CallbackExecutor.cpp

```diff
@@ -31,18 +31,18 @@ CallbackExecutor::CallbackExecutor(function<void(function<void()> call)> callbac
                 }
 
                 lock.unlock();
-                for (const auto& p : queue)
+                for (const auto& callback : queue)
                 {
                     try
                     {
                         if (_callbackExecutor)
                         {
-                            // TODO do we need to ensure p.second is executed before we continue?
-                            _callbackExecutor(p.second);
+                            // TODO do we need to ensure that the callback is executed before we continue?
```

Comment:
I don't think we do and I don't see how we would do it.

Reply:
> I don't think we do

I guess destroy just cares about joining the thread; it just so happens that, without the custom executors, this also means all callbacks have finished executing.

>  I don't see how we would do it.

Wrapping the callback in another callback, the wrapper sets a condition variable, and the thread sending callbacks to the custom executor waits for the condition.